multi-int-store/src/main/java/com/connexta/multiintstore/adaptors/S3StorageAdaptor.java

```diff
@@ -67,30 +75,29 @@ public void store(
         key);
   }
 
-  // TODO: Uncomment when the MIS implements retrieval
+  @Override
+  public RetrieveResponse retrieve(String key) throws StorageException {
+    log.info("Retrieving product in bucket \"{}\" with key \"{}\"", s3BucketQuarantine, key);
 
-  //  @Override
-  //  public RetrieveResponse retrieve(String ingestId) throws IOException {
-  //    log.info(
-  //        "Retrieving product in bucket \"{}\" with key \"{}\"", s3BucketQuarantine, ingestId);
-  //
-  //    final ResponseInputStream<GetObjectResponse> getObjectResponseResponseInputStream =
-  //        s3Client.getObject(
-  //            GetObjectRequest.builder().bucket(s3BucketQuarantine).key(ingestId).build());
-  //    final GetObjectResponse getObjectResponse = getObjectResponseResponseInputStream.response();
-  //
-  //    return new RetrieveResponse(
-  //        MediaType.valueOf(getObjectResponse.contentType()),
-  //        new ByteArrayResource(getObjectResponseResponseInputStream.readAllBytes()) {
-  //          @Override
-  //          public String getFilename() {
-  //            final Map<String, String> metadata = getObjectResponse.metadata();
-  //            final String filename = metadata.get("filename");
-  //            if (StringUtils.isEmpty(filename)) {
-  //              return ingestId;
-  //            }
-  //            return filename;
-  //          }
-  //        });
-  //  }
+    try (final ResponseInputStream<GetObjectResponse> getObjectResponseResponseInputStream =
+        s3Client.getObject(
+            GetObjectRequest.builder().bucket(s3BucketQuarantine).key(key).build())) {
+      final GetObjectResponse getObjectResponse = getObjectResponseResponseInputStream.response();
+      return new RetrieveResponse(
+          MediaType.valueOf(getObjectResponse.contentType()),
+          new ByteArrayResource(getObjectResponseResponseInputStream.readAllBytes()) {
+            @Override
+            public String getFilename() {
```

Comment:
:exclamation: : The getFilename() method is overriden to return a Filename or a key. It could be confusing to ask for a filename and not know if the method will return an actual filename or something that was not the filename. Are they cases where a called would want either the filename or the key and not care which as which?

Also, are there situations where the filename can be unknown?

Reply:
I updated RetrieveResponse to include a `fileName`. All of the products in S3 should have a fileName metadata value, so I updated S3StorageAdaptor to throw an error if the name isn't there instead of defaulting to the key.